smartsim/entity/ensemble.py

```diff
@@ -59,23 +59,182 @@ def __init__(
         max_permutations: int = -1,
         replicas: int = 1,
     ) -> None:
+        """Initialize an ``Ensemble`` of application instances
+
+        :param name: name of the ensemble
+        :param exe: executable to run
+        :param exe_args: executable arguments
+        :param exe_arg_parameters: parameters and values to be used when configuring entities
+        :param files: files to be copied, symlinked, and/or configured prior to
+                      execution
+        :param file_parameters: parameters and values to be used when configuring
+                                files
+        :param permutation_strategy: strategy to control how the param values are applied to the Ensemble
+        :param max_permutations: max parameter permutations to set for the ensemble
+        :param replicas: number of identical entities to create within an Ensemble
+        :return: and ``Ensemble`` instance
+        """
         self.name = name
-        self.exe = os.fspath(exe)
-        self.exe_args = list(exe_args) if exe_args else []
-        self.exe_arg_parameters = (
+        """The name of the ensemble """
+        self._exe = os.fspath(exe)
+        """The executable to run """
+        self._exe_args = list(exe_args) if exe_args else []
+        """The executable arguments """
```

Comment:
Nit: They should get stripped out if/when we generate docs, but I tend to avoid the trailing spaces in the docstrings
```suggestion
        """The name of the ensemble"""
        self._exe = os.fspath(exe)
        """The executable to run"""
        self._exe_args = list(exe_args) if exe_args else []
        """The executable arguments"""
```

Reply:
removed whitespace